apimodels/cedar_models.go

```diff
@@ -1,9 +1,155 @@
+	"github.com/mongodb/grip"
+	"github.com/mongodb/grip/level"
+	"github.com/pkg/errors"
+)
+
 type CedarConfig struct {
 	BaseURL  string `json:"base_url"`
 	RPCPort  string `json:"rpc_port"`
 	Username string `json:"username"`
 	APIKey   string `json:"api_key,omitempty"`
 	Insecure bool   `json:"insecure"`
 }
+
+// GetBuildloggerLogsOptions represents the arguments passed into the
+// GetBuildloggerLogs function.
+// TODO (EVG-20019): Remove the "V2" suffix once log fetching is consolidated
+// and the original GetBuildloggerLogsOptions struct is deleted.
+type GetBuildloggerLogsOptionsV2 struct {
+	BaseURL   string   `json:"_"`
+	TaskID    string   `json:"-"`
+	Execution *int     `json:"-"`
+	TestName  string   `json:"-"`
+	Tags      []string `json:"-"`
+	Start     int64    `json:"-"`
+	End       int64    `json:"-"`
+	Limit     int      `json:"-"`
+	Tail      int      `json:"-"`
+}
+
+// GetBuildloggerLogs makes request to Cedar for a specifc log and returns a
+// log iterator.
+// TODO (EVG-20019): Remove the "V2" suffix once log fetching is consolidated
+// and the original GetBuildloggerLogs function is deleted.
+func GetBuildloggerLogsV2(ctx context.Context, opts GetBuildloggerLogsOptionsV2) (log.LogIterator, error) {
+	usr := gimlet.GetUser(ctx)
+	if usr == nil {
+		return nil, errors.New("error getting user from context")
+	}
+
+	var start, end time.Time
+	if opts.Start > 0 {
+		start = time.Unix(0, opts.Start).UTC()
+	}
+	if opts.End > 0 {
+		end = time.Unix(0, opts.Start).UTC()
+	}
+
+	getOpts := buildlogger.GetOptions{
+		Cedar: timber.GetOptions{
+			BaseURL:  fmt.Sprintf("https://%s", opts.BaseURL),
+			UserKey:  usr.GetAPIKey(),
+			UserName: usr.Username(),
+		},
+		TaskID:        opts.TaskID,
+		Execution:     opts.Execution,
+		TestName:      opts.TestName,
+		Tags:          opts.Tags,
+		PrintTime:     true,
+		PrintPriority: true,
+		Start:         start,
+		End:           end,
+		Limit:         opts.Limit,
+		Tail:          opts.Tail,
+	}
+	r, err := buildlogger.Get(ctx, getOpts)
+	if err != nil {
+		return nil, errors.Wrapf(err, "getting logs from Cedar Buildlogger")
+	}
+
+	return newBuildloggerIterator(r), nil
+}
+
+type buildloggerIterator struct {
+	readCloser io.ReadCloser
+	reader     *bufio.Reader
+	item       log.LogLine
+	catcher    grip.Catcher
+	exhausted  bool
+	closed     bool
+}
+
+func newBuildloggerIterator(r io.ReadCloser) *buildloggerIterator {
+	return &buildloggerIterator{
+		readCloser: r,
+		reader:     bufio.NewReader(r),
+		catcher:    grip.NewBasicCatcher(),
+	}
+}
+
+func (it *buildloggerIterator) Next() bool {
+	if it.closed || it.exhausted {
+		return false
+	}
+
+	line, err := it.reader.ReadString('\n')
+	if err != nil {
+		it.exhausted = err == io.EOF
+		it.catcher.AddWhen(err != io.EOF, errors.Wrap(err, "reading log lines"))
+		return false
+	}
+
+	// Parse prefixed priority with format "[P:3%d] %s".
+	priority, err := strconv.Atoi(strings.TrimSpace(line[3:6]))
```

Comment:
Are we confident that `line` will definitely have this length, and there's no way to get a log line that won't have this format? Otherwise it'll panic.

If we're not 100% confident that the log lines will always be formatted like this, I think we should length check it, and also verify that the log line follows the expected format.

Reply:
Good idea! Added a check above that will fail the iterator if the line does not have at least 34 characters.